### ISSUE TITLE
Support train/test flag in evaluator

### ISSUE THREAD

**Issue by ryotat** [pfnet/chainer#2102]
I can see in both CIFAR and ImageNet examples Evaluator is extended like this
```python
class TestModeEvaluator(extensions.Evaluator):
    def evaluate(self):
        model = self.get_target('main')
        model.train = False
        ret = super(TestModeEvaluator, self).evaluate()
        model.train = True
        return ret
```
Shouldn't this be done by default? Or at least the documentation should mention that this is necessary. It is hard to imagine a network without dropout or batch normalization these days and this could be a very easy pitfall.

It used to be the case that the users take care whether the model should run in the training mode or in the evaluation mode but now most users would think it is the framework's responsibility to take care of this.

**Comment by beam2d** [comment]
You are right, the train/test phase configuration should be managed by frameworks. I'm planning to introduce a centralized way to configure train/test mode as a part of general configuration interface #2048, and it should include such an automatic configuration that the framework takes care of.

**Comment by unnonouno** [comment]
In v2, you can use train-mode flag. Please check it.
